llm_toolkit/models.py

```diff
@@ -263,6 +272,41 @@ class GPT4o(GPT):
   name = 'gpt-4o'
 
 
+class AzureGPT(GPT):
+  """Azure's GPT model."""
+
+  name = 'gpt-3.5-turbo-azure'
+
+  def _get_tiktoken_encoding(self):
+    """Returns the tiktoken encoding for the model."""
+    try:
+      return tiktoken.encoding_for_model(self.name.replace('-azure', ''))
+    except KeyError:
+      logger.info('Could not get a tiktoken encoding for %s.', self.name)
+      return tiktoken.get_encoding('cl100k_base')
```

Comment:
Very minor nit: We can also make the method receive as argument the name of the model (in the parent class), then this one would just be

```
super()._get_tiktoken_encoding(self.name.replace('-azure', ''))
```

and the rest of the lines would be handled by the parent class.

But let's leave this for future work